src/Kit/System/Watchdog/RawThread.h

```diff
@@ -0,0 +1,137 @@
+#ifndef KIT_SYSTEM_WATCHDOG_RAWTHREAD_H_
+#define KIT_SYSTEM_WATCHDOG_RAWTHREAD_H_
+/*------------------------------------------------------------------------------
+ * Copyright Integer Fox Authors
+ *
+ * Distributed under the BSD 3 Clause License. See the license agreement at:
+ * https://github.com/Integerfox/kit.core/blob/main/LICENSE
+ *
+ * Redistributions of the source code must retain the above copyright notice.
+ *----------------------------------------------------------------------------*/
+/** @file */
+
+#include "kit_config.h"
+#include "WatchedThread.h"
+#include "Supervisor.h"
+#include <cstdint>
+
+///
+namespace Kit {
+///
+namespace System {
+///
+namespace Watchdog {
+
+// ============================================================================
+// Watchdog Macros for Raw Threads
+// ============================================================================
+
+/** Compile-time macros for instrumenting raw threads with watchdog monitoring.
+    These macros can be enabled/disabled at compile time and support both
+    direct function calls and pointer-based calls for flexibility.
+ */
+
+#ifdef USE_KIT_SYSTEM_WATCHDOG
+
+/** Macro to start watchdog monitoring for a raw thread.
+    Usage: KIT_SYSTEM_WATCHDOG_START_RAWTHREAD(threadPtr)
+
+    @param threadPtr The RawThread instance pointer
+ */
+#define KIT_SYSTEM_WATCHDOG_START_RAWTHREAD( threadPtr ) \
+    do                                                   \
+    {                                                    \
+        if ( threadPtr )                                 \
+        {                                                \
+            threadPtr->startWatching();                  \
+        }                                                \
+    }                                                    \
+    while ( 0 )
```

Comment:
You do not need to pass a pointer for the Raw thread.  It was necessary on the WatchedEventThread because enabling/using a Watchdog with an event loop is an option feature - and the watched thread point would be null if not using the Watchdog
```suggestion
#define KIT_SYSTEM_WATCHDOG_START_RAWTHREAD( instance )  instance.startWatching()
```